### DISCUSSION TITLE
Got an error when runnin "UFS_IC_generator.py" (CCPP-SCM v5.0.0)

### DISCUSSION THREAD

**Discussion by wzzheng90** [NCAR/ccpp-scm#315] [Cases, Initial Conditions, Forcing] [answered]
Hi,

Within the scm/etc/script, I tried "UFS_IC_generator.py" script to generate the example UFS Atmosphere initial condition case input file.

I got an error when I run "UFS_IC_generator.py" to create ICs on NOAA's Hera HPC. For example, I just try a sample case as follows:

./UFS_IC_generator.py -l 261.51 38.2 -d 201610030000 -i ../../data/raw_case_input/FV3_C96_example_ICs -g ../../data/raw_case_input/FV3_C96_example_ICs -n fv3_model_point_noah -oc

The error message shows as follows:
File "./UFS_IC_generator.py", line 2045
print 'Tile found: {0}'.format(tile)
^
SyntaxError: invalid syntax

BTW, the shapely python package is installed on Hera.  Is it still a problem related to python version?  

Thanks,

Weizhong

**Comment by dustinswales** [comment]
@wzzheng90 I can replicate this error on Hera. It may be that the setup Hera setup script needs updating. FWIW, this script works on Cheyenne. I will take a look this afternoon.

**Comment by dustinswales** [comment, marked as answer]
In the meantime you can use this environment to run the script.
conda create --name py27env --clone /scratch2/BMC/ome/Dustin.Swales/libs/miniconda3/envs/py27
conda activate py27env

**Reply by wzzheng90** [reply]
@dustinswales :  Many thanks!!

       I tried to use this environment according to your suggestion. Then, I run the script.  Another problem happens as follows:
===================================
Traceback (most recent call last):
  File "./UFS_IC_generator.py", line 7, in <module>
    from netCDF4 import Dataset
ImportError: No module named netCDF4
=====================================  

    Line 7 in "UFS_IC_generator.py":          from netCDF4 import Dataset
How to fix this problem?

Wezhong

**Reply by dustinswales** [reply]
> 

@wzzheng90 
I think if you repeat this process, but for py37_scm, it should work.

**Reply by dustinswales** [reply]
> 

@wzzheng90 
Sorry, I just restored my py27 environment. You should be able to clone this as before and it "should" work.

**Reply by wzzheng90** [reply]
@dustinswales   Really appreciate it.  It's working now.  Thanks again.

**Comment by dustinswales** [comment]
Make sure that you are using the correct python.
![Screen Shot 2022-05-17 at 2 54 57 PM](https://user-images.githubusercontent.com/13949033/168908610-9a24eb2c-4646-4616-8055-f2b36be33875.png)
:

**Comment by dustinswales** [comment]
[Dustin.Swales@hfe09 scripts]$ where python
/usr/bin/python
[Dustin.Swales@hfe09 scripts]$ conda activate py27env
(py27env) [Dustin.Swales@hfe09 scripts]$ where python
**/scratch2/BMC/ome/Dustin.Swales/libs/miniconda3/envs/py27env/bin/python**
/usr/bin/python

You should see a second python that appears when you activate py27env?

**Reply by wzzheng90** [reply]
I tried again.  It does not see a second python:

%where python
/usr/bin/python

%conda activate py27env
%where python
/usr/bin/python

**Reply by wzzheng90** [reply]
I tried it again.    I did not see the second python, but see (py27env) environment?

zeuse /scratch2/NCEPDEV/land/Weizhong.Zheng/noscrub 257 %where python
/usr/bin/python
zeuse /scratch2/NCEPDEV/land/Weizhong.Zheng/noscrub 258 %conda activate py27env
(py27env) zeuse /scratch2/NCEPDEV/land/Weizhong.Zheng/noscrub 259 %where python
/usr/bin/python
(py27env) zeuse /scratch2/NCEPDEV/land/Weizhong.Zheng/noscrub 260 %

**Comment by wzzheng90** [comment]
@dustinswales   It works now if "/scratch2/BMC/ome/Dustin.Swales/libs/miniconda3/envs/py27env/bin/python" is used before UFS_IC_generator.py,  although I can not see the second python if I type "where python".

Thank you very much for your kind help !!

**Reply by dustinswales** [reply]
Great that it's working!

**Reply by wzzheng90** [reply]
@dustinswales  I use your "/scratch2/BMC/ome/Dustin.Swales/libs/miniconda3/envs/py27env/bin/python" for UFS_IC_generator.py.
However, it looks like "py27env" was removed, and you updated something.  Could you please inform me how to update?  Thanks.

In the past,  I used  "conda activate py27env".

**Reply by wzzheng90** [reply]
@dustinswales   I did as follows but it still got a problem.

% conda create --name py37_scmenv --clone /scratch2/BMC/ome/Dustin.Swales/libs/miniconda3/envs/py37_scm
% conda activate py37_scmenv
%where python
/home/Weizhong.Zheng/.conda/envs/py37_scmenv/bin/python
/usr/bin/python

% /home/Weizhong.Zheng/.conda/envs/py37_scmenv/bin/python ./UFS_IC_generator.py  .......

  File "./UFS_IC_generator.py", line 2045
    print 'Tile found: {0}'.format(tile)
                          ^
SyntaxError: invalid syntax

It looks like I still not use the correct python version?

Thanks.

**Reply by wzzheng90** [reply]
@dustinswales   I tried "py37_scm" again but still not working.  Do you still keep "py27env" on your directory? Thanks.

**Comment by wzzheng90** [comment]
Really appreciate it.   It's working now.

On Wed, Jun 22, 2022 at 6:40 PM dustinswales ***@***.***>
wrote:

> @wzzheng90 <https://github.com/wzzheng90>
> Sorry, I just restored my py27 environment. You should be able to clone
> this as before and it "should" work.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NCAR/ccpp-scm/discussions/315#discussioncomment-3005837>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALOCL25N2AEC5I7FBC7H66LVQOI4NANCNFSM5WFZQPQA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Weizhong Zheng, Dr.
IMSG at Environmental Modeling Center
NCEP/NWS/NOAA
5830 University Research Court , #2028
College Park, MD 20740
TEL: 301-683-3694 (O)
FAX: 301-683-3703
Email: ***@***.*** ***@***.***>

**Comment by christianbsch** [comment]
Hello, 

I am having the same problem with the UFS_IC_generator.py (as shown by wwzhen90 in the first part of this question), but in Cheyenne. There is any specific conda environment that I should use at Cheyenne? or there is any updated version of this script (in Python 3.0+)?

Regards

Christian

**Reply by hertneky** [reply]
@christianbsch This issue is fixed in the main branch. Please refer to the documentation (ccpp-scm/scm/doc/TechGuide/main.pdf) in section 5.5 for setting up the environment and running the script.
